### PR TITLE
Tic-tac-toe command

### PR DESCRIPTION
**Pull request by Shivansh-007**
## Description
`!ttt` or `!tictactoe` command lets users play tic-tac-toe using reactions, either player vs player or player vs AI-like program. Allowing them to select their own tokens (default 🍉 and 🥒 )

Thanks to ks123's lovely tic-tac-toe pr on lancebot, i could improve mine (had been using a `while` loop instead of just looping 9 times and had a long check while after looking ks123, it is a simple hard coded check

## Screenshots
Screenshot from 2021-02-01 05-43-40
Screenshot from 2021-02-01 05-43-47
Screenshot from 2021-02-01 05-43-59

### CODE REVIEW

File: bot/exts/fun/tic_tac_toe.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+        )
+        self.games.append(game)
+        await game.play()
+
+    @tic_tac_toe.group(aliases=["bot", "computer", "cpu"])
+    async def ai(self, ctx: Context, emoji1: EMOJI_CHECK = Emojis.cucumber) -> None:
+        """Play tic-tac-toe against a computer player."""
+        if isinstance(emoji1, str) and len(emoji1) > 1:
+            raise discord.ext.commands.EmojiNotFound(emoji1)
+        game = Game([Player(ctx.author, ctx, str(emoji1)), AI(Emojis.watermelon)], ctx)
```
Comment (vivax3794) [inline bot/exts/fun/tic_tac_toe.py:358]: here you are not using the passed in emoji

File: bot/exts/fun/tic_tac_toe.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+def is_requester_channel_free() -> t.Callable:
+    """Check is channel where command will be invoked free and if the requester is free."""
+
+    async def predicate(ctx: Context) -> bool:
+        channel_free = all(
+            game.channel != ctx.channel for game in ctx.cog.games if not game.over
+        )
+
+        requester_free = all(
+            ctx.author not in (player.user for player in game.players if player != "AI")
```
Comment (vivax3794) [inline bot/exts/fun/tic_tac_toe.py:230]: `player` would be an `AI` object right, not a string?

File: bot/exts/fun/tic_tac_toe.py (hunk 1 above)
Comment (Shivansh-007) [reply bot/exts/fun/tic_tac_toe.py:358]:
```
!tictactoe [ai|bot|computer|cpu] [emoji1=🥒]
```
It only accepts one emoji.

File: bot/exts/fun/tic_tac_toe.py (hunk 2 above)
Comment (Shivansh-007) [reply bot/exts/fun/tic_tac_toe.py:230]: Yep, will fix that.

**Comment by Shivansh-007** [comment]
Closing PR as I am not interested in contributing to the bot anymore, if anyone wants to take over you can open a new PR.

**Pull request closed without merging**